lib/Form/Type/DateTimeIntervalType.php

```diff
@@ -0,0 +1,67 @@
+<?php
+/**
+ * This file is part of the eZ RepositoryForms package.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ * @version //autogentag//
+ */
+
+namespace EzSystems\RepositoryForms\Form\Type;
+
+use EzSystems\RepositoryForms\Form\DataTransformer\DateIntervalToArrayTransformer;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\ReversedTransformer;
+use Symfony\Component\Form\Extension\Core\DataTransformer\DataTransformerChain;
+use Symfony\Component\Form\Extension\Core\DataTransformer\ArrayToPartsTransformer;
+use Symfony\Component\OptionsResolver\OptionsResolverInterface;
+
+/**
+ * Form type for ContentType update.
+ */
+class DateTimeIntervalType extends AbstractType
+{
+    public function getParent()
+    {
+        return 'form';
+    }
+
+    public function getName()
+    {
+        return 'datetimeinterval';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $parts = array( 'year', 'month', 'day', 'hour', 'minute', 'second' );
+        $dateParts = array( 'year', 'month', 'day' );
+        $timeParts = array( 'hour', 'minute', 'second' );
+
+        $builder->addViewTransformer(
+            new DataTransformerChain(
+                array(
+                    new DateIntervalToArrayTransformer($parts),
+                    new ArrayToPartsTransformer(
+                        array(
+                            'date' => $dateParts,
+                            'time' => $timeParts,
+                        )
+                    ),
+                )
+            )
+        )
+            ->add('date', 'date')
```

Comment:
This is what phpstorm wants. Looks weird yes, but what is correct - 4 spaces back?